### PR TITLE
Pack relocations for Android API >= 28

### LINKED ISSUE #909: [BUG] Hoist relocation packing flags into the Clang driver
These aren't useful until `minSdkVersion >= 28`, but we should hoist the `-Wl,--pack-dyn-relocs=android+relr` and `-Wl,--use-android-relr-tags` into the driver. So they are used automatically when available.

### PR DESCRIPTION
**Pull request by hiraditya**
Patch copied from: https://github.com/android/ndk/issues/909#issuecomment-649872696
Fixes: https://github.com/android/ndk/issues/909

### CODE REVIEW

File: clang/lib/Driver/ToolChains/Linux.cpp [hunk 1]
```diff
@@ -256,6 +256,14 @@ Linux::Linux(const Driver &D, const llvm::Triple &Triple, const ArgList &Args)
       ExtraOpts.push_back("-z");
       ExtraOpts.push_back("max-page-size=16384");
     }
+    if (Triple.isAndroidVersionLT(29)) {
```
Comment (enh-google) [inline clang/lib/Driver/ToolChains/Linux.cpp:259]: this says "<" but the commit message says "<="?

File: clang/lib/Driver/ToolChains/Linux.cpp [hunk 2]
```diff
@@ -256,6 +256,14 @@ Linux::Linux(const Driver &D, const llvm::Triple &Triple, const ArgList &Args)
...
+      // https://github.com/android/ndk/issues/1196
+      // The unwinder used by the crash handler on versions of Android prior to
```
Comment (enh-google) [inline clang/lib/Driver/ToolChains/Linux.cpp:261]: ah, sounds like the commit message is wrong then? (or at least "outvoted" :-) )

File: clang/lib/Driver/ToolChains/Linux.cpp (hunk 1 above)
Comment (hiraditya) [reply clang/lib/Driver/ToolChains/Linux.cpp:259]: Edited the commit message

Comment (enh-google) [reply clang/lib/Driver/ToolChains/Linux.cpp:259]: i meant "Add no-rosegment by default for API 29 and earlier." --- i actually hadn't noticed the _pull request_ message. i think that was already right? (or at least, "consistent with the code and code comments".)

**Review by enh-google — APPROVED** [review]

**Review by DanAlbert — APPROVED** [review]

**Comment by vitalybuka** [comment]
breaks this bot https://lab.llvm.org/buildbot/#/builders/186/builds/4581

File: clang/lib/Driver/ToolChains/Linux.cpp [hunk 3]
```diff
@@ -256,6 +256,24 @@ Linux::Linux(const Driver &D, const llvm::Triple &Triple, const ArgList &Args)
...
+      // API 29 did not correctly handle binaries built with rosegment, which is
+      // enabled by default for LLD. Android only supports LLD, so it's not an
```
Comment (glandium) [inline clang/lib/Driver/ToolChains/Linux.cpp:263]: The NDK only provides LLD, but one can still use another linker via -fuse-ld, and this breaks that use case. And yes, there are people doing that.

**Pull request merged**